qt/aqt/ankihub.py

```diff
@@ -0,0 +1,164 @@
+
+    def callback(username: str, password: str) -> None:
+        if not username and not password:
+            return
+        if username and password:
+            mw.taskman.with_progress(
+                lambda: mw.col.ankihub_login(id=username, password=password),
+                functools.partial(on_future_done, username=username, password=password),
+                parent=mw,
+            )
+        else:
+            ankihub_login(mw, on_success, username, password, from_prefs_screen)
+
+    get_id_and_pass_from_user(mw, callback, username, password, from_prefs_screen)
+
+
+def ankihub_logout(
+    mw: aqt.main.AnkiQt,
+    on_success: Callable[[], None],
+    token: str,
+) -> None:
+    mw.taskman.with_progress(
+        lambda: mw.col.ankihub_logout(token=token),
+        # We don't need to wait for the response
+        lambda _: on_success(),
+        parent=mw,
+    )
+
+
+def get_id_and_pass_from_user(
+    mw: aqt.main.AnkiQt,
+    callback: Callable[[str, str], None],
+    username: str = "",
+    password: str = "",
+    from_prefs_screen: bool = False,
+) -> None:
+    diag = QDialog(mw)
+    diag.setWindowTitle("Anki")
+    disable_help_button(diag)
+    diag.setWindowModality(Qt.WindowModality.WindowModal)
+    diag.setMinimumWidth(600)
+    vbox = QVBoxLayout()
+    info_label = QLabel(
+        f"<h1>{tr.sync_ankihub_dialog_heading()}</h1>{tr.preferences_ankihub_intro()}"
+    )
+    info_label.setOpenExternalLinks(True)
+    info_label.setWordWrap(True)
+    vbox.addWidget(info_label)
+    vbox.addSpacing(20)
+    g = QGridLayout()
+    l1 = QLabel(tr.sync_ankihub_username_label())
+    g.addWidget(l1, 0, 0)
+    user = QLineEdit()
+    user.setText(username)
+    g.addWidget(user, 0, 1)
+    l2 = QLabel(tr.sync_password_label())
+    g.addWidget(l2, 1, 0)
+    passwd = QLineEdit()
+    passwd.setText(password)
+    passwd.setEchoMode(QLineEdit.EchoMode.Password)
+    g.addWidget(passwd, 1, 1)
+    vbox.addLayout(g)
+
+    vbox.addSpacing(20)
+    sign_in_label = QLabel(
+        f"{tr.sync_ankihub_signup_prompt()} "
+        f'<a href="https://app.ankihub.net/accounts/signup/">{tr.sync_sign_up()}</a><br>'
+        f'<a href="https://app.ankihub.net/accounts/password/reset/">{tr.sync_forgot_password()}</a>'
+    )
+    sign_in_label.setOpenExternalLinks(True)
+    sign_in_label.setWordWrap(True)
+    vbox.addWidget(sign_in_label)
+
+    vbox.addSpacing(20)
+    bb = QDialogButtonBox()  # type: ignore
+    sign_in_button = QPushButton(tr.sync_sign_in())
+    sign_in_button.setAutoDefault(True)
+    bb.addButton(
+        QPushButton(tr.actions_cancel() if from_prefs_screen else tr.actions_skip()),
+        QDialogButtonBox.ButtonRole.RejectRole,
+    )
+    bb.addButton(
+        sign_in_button,
+        QDialogButtonBox.ButtonRole.AcceptRole,
+    )
+    qconnect(bb.accepted, diag.accept)
+    qconnect(bb.rejected, diag.reject)
+    vbox.addWidget(bb)
+
+    diag.setLayout(vbox)
+    diag.show()
```

Comment:
:wrench: When I opened the AnkiHub login dialog, the text was cut off:
<img src="https://github.com/user-attachments/assets/99c869d1-b8da-4b33-8e2e-6838f52fb7f8" width="500" />

```suggestion
    diag.adjustSize()
    diag.show()
```

AnkiWeb's login dialog has the same problem:
https://github.com/abdnh/anki/blob/1ef83f07f61b178334eb5ce8710648d6bd1c34a3/qt/aqt/sync.py#L339

<img src="https://github.com/user-attachments/assets/05de2b06-17b6-4cd8-b8cf-af292f0f908c" width="200" />
<br><br>
After adding `diag.adjustSize()`:<br>
<img src="https://github.com/user-attachments/assets/05230a86-a224-40ba-9744-df034a3c2295" width="500" />